web/app/js/components/NamespaceLanding.jsx

```diff
@@ -0,0 +1,182 @@
+    }).isRequired,
+    controllerNamespace: PropTypes.string.isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.api = this.props.api;
+    this.handleApiError = this.handleApiError.bind(this);
+    this.loadFromServer = this.loadFromServer.bind(this);
+    this.state = this.getInitialState();
+  }
+
+  getInitialState() {
+    return {
+      selectedNs: null,
+      metricsByNs: {},
+      namespaces: [],
+      pollingInterval: 2000,
+      pendingRequests: false,
+      loaded: false,
+      error: null
+    };
+  }
+
+  componentDidMount() {
+    this.loadFromServer();
+    this.timerId = window.setInterval(this.loadFromServer, this.state.pollingInterval);
+  }
+
+  componentWillUnmount() {
+    window.clearInterval(this.timerId);
+    this.api.cancelCurrentRequests();
+  }
+
+  onNsSelect = ns => {
+    this.setState({
+      selectedNs: ns
+    });
+  }
+
+  loadFromServer() {
+    if (this.state.pendingRequests) {
+      return; // don't make more requests if the ones we sent haven't completed
+    }
+    this.setState({ pendingRequests: true });
+
+    let apiRequests = [
+      this.api.fetchMetrics(this.api.urlsForResource("namespace"))
+    ];
+    if (!_.isEmpty(this.state.selectedNs)) {
+      apiRequests = _.concat(apiRequests, [this.api.fetchMetrics(this.api.urlsForResource("all", this.state.selectedNs))]);
+    }
+    this.api.setCurrentRequests(apiRequests);
+
+    Promise.all(this.api.getCurrentPromises())
+      .then(([allNs, metricsForNs]) => {
+        let namespaces = processSingleResourceRollup(allNs);
+        let metricsByNs = this.state.metricsByNs;
+        if (!_.isNil(this.state.selectedNs) && !_.isNil(metricsForNs)) {
+          metricsByNs[this.state.selectedNs] = processMultiResourceRollup(metricsForNs);
+        }
+
+        // by default, show the first non-linkerd meshed resource
+        let defaultOpenNs = _.find(namespaces, ns => ns.added && ns.name !== this.props.controllerNamespace);
+
+        this.setState({
+          namespaces,
+          metricsByNs,
+          defaultOpenNs,
+          selectedNs: this.state.selectedNs || defaultOpenNs.name,
+          pendingRequests: false,
+          loaded: true,
+          error: null
+        });
+      })
+      .catch(this.handleApiError);
+  }
+
+  handleApiError = e => {
+    if (e.isCanceled) {
+      return;
+    }
+
+    this.setState({
+      pendingRequests: false,
+      error: e
+    });
+  }
+
+  renderResourceSection(resource, metrics) {
+    if (_.isEmpty(metrics)) {
+      return null;
+    }
+    return (
+      <div className="page-section">
+        <h3>{friendlyTitle(resource).plural}</h3>
+        <MetricsTable
+          resource={resource}
+          metrics={metrics}
+          showNamespaceColumn={false} />
+      </div>
+    );
+  }
+
+  renderNamespaceSection(namespace) {
+    let metrics = this.state.metricsByNs[namespace] || {};
+    let noMetrics = _.isEmpty(metrics.pod);
```

Comment:
Related to @dadjeibaah suggestion, I think we should distinguish between metrics not loaded and no actual metrics by displaying a spinner until the `metricsByNs` map has a key for the given namespace. Maybe something like this would work:

```
    if (!_.has(this.state.metricsByNs, namespace)) {
      return <Spin size="large" />;
    }
    let metrics = this.state.metricsByNs[namespace];
    let noMetrics = _.isEmpty(metrics.pod);
```

Reply:
ooh, yeah, nice solution!